frappe/desk/form/load.py

```diff
@@ -43,7 +43,8 @@ def getdoc(doctype, name, user=None):
 		)
 		raise frappe.PermissionError(("read", doctype, name))
 
-	doc.apply_fieldlevel_read_permissions()
+	if frappe.get_system_settings("apply_perm_level_on_api_calls"):
```

Comment:
This is one place that was already patched. Applying permlevel filters on the accessed Document unconditionally in this `getdoc` API causes inconsistencies with the `frappe.api` & `frappe.client` endpoints.

Should all the Document APIs apply the patch unconditionally, and only the setting only apply the restriction for DatabaseQuery? Or, we apply this unconditionally just here and keep the APIs' behaviour inconsistent but non-breaking? Or, something else?

Reply:
Mediated by not changing behaviour for desk API but applying new behaviour on other APIs